nilearn/plotting/matrix_plotting.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.tight_layout import get_renderer
+from matplotlib.colorbar import make_axes
+
+
+def fit_axes(ax):
+    """ Redimension the given axes to have labels fitting.
+    """
+    fig = ax.get_figure()
+    renderer = get_renderer(fig)
+    ylabel_width = ax.yaxis.get_tightbbox(renderer).inverse_transformed(
+        ax.figure.transFigure).width
+    if ax.get_position().xmin < 1.1 * ylabel_width:
+        # we need to move it over
+        new_position = ax.get_position()
+        new_position.x0 = 1.1 * ylabel_width  # pad a little
+        ax.set_position(new_position)
+
+    xlabel_height = ax.xaxis.get_tightbbox(renderer).inverse_transformed(
+        ax.figure.transFigure).height
+    if ax.get_position().ymin < 1.1 * xlabel_height:
+        # we need to move it over
+        new_position = ax.get_position()
+        new_position.y0 = 1.1 * xlabel_height  # pad a little
+        ax.set_position(new_position)
+
+
+def plot_matrix(mat, ax=None, tri='full', title=None, labels=None,
+                auto_fit=True, grid=False, colorbar=True,
+                **kwargs):
+    """ Plot the given matrix.
+
+        Parameters
+        ----------
+        mat : 2-D numpy array
+            Matrix to be plotted.
+        ax  : None or Axes.
+            Axes instance to be plotted on. Creates a new one if None.
+        tri : {'lower', 'diag', 'full'}
+            Which triangular part of the matrix to plot:
+            'lower' is the lower part, 'diag' is the lower including
+            diagonal, and 'full' is the full matrix.
+        text : string or None
+            A text to add in the upper left corner.
+        labels : list of strings
+            The label of each row and column
+        auto_fit : boolean, optional
+            If auto_fit is True, the axes are dimensioned to give room
+            for the labels. This assumes that the labels are resting
+            against the bottom and left edges of the figure.
+        grid : color or False
+            If not False, a grid is plotted to separate rows and columns
+            using the given color.
```

Comment:
Almost all the examples use "cmap=RdBu_r". Maybe we should add a cmap argument, with the default "RdBu_r", to facilitate things for our users, and pass it along to imshow.